### PR TITLE
Follow up fixes from TLSVERIFY work

### PR DESCRIPTION
**Pull request by hickeng**
Block vic-machine upgrade from completing until the components
have come up cleanly. This does **not** add rollback if the VCH fails
to initialize correctly.

Only serves the docker API on the client interface as designed and
removes some of the unnecessary flags.

Updates to examples and usage doc.

Towards #2624 (handles the --no-tls case)
Fixes: #2197

### CODE REVIEW

**Comment by hickeng** [comment]
This is now rather overloaded and in need of me cherry picking into different PRs.

File: cmd/docker/main.go [hunk 1]
```diff
@@ -222,14 +191,38 @@ func startServerWithOptions(cli *CliOptions) *apiserver.Server {
 		}
 	}
 
+	addr := "0.0.0.0"
+	// exposing this on all interfaces
+	if vchConfig.Diagnostics.DebugLevel <= 2 {
```
Comment (andrewtchin) [inline cmd/docker/main.go]: Why does the lookup depend on the debug level?

File: cmd/vic-init/ops.go [hunk 2]
```diff
@@ -69,6 +70,9 @@ func (t *operations) HandleSessionExit(config *tether.ExecutorConfig, session *t
 			return
 		}
 
+		// incredibly basic throttle
```
Comment (hmahmood) [inline cmd/vic-init/ops.go]: Please tag this as a TODO. This indicates that we have issues around synchronization; we should not be putting in random sleeps.

Comment (hmahmood) [inline cmd/vic-init/ops.go]: Actually why do we have this? I don't think this is a good idea.

**Comment by hickeng** [comment]
Split this into multiple PRs:
#2703 
#2706 
#2707 

and there'll be another one for limiting the servers to listening on appropriate interfaces & #2632

**Pull request closed without merging**